### ISSUE TITLE
Amesos2: Randomly failing test

### ISSUE THREAD

**Issue by cgcgcg** [trilinos/Trilinos#11454]
@trilinos/amesos2 @srajama1 @ndellingwood 

## Bug Report
On Cuda with UVM=OFF I get random failures from `Amesos2_CrsMatrix_Adapter_Consistency_Tests_MPI_4`. The problematic unit test is `CrsMatrixAdapter_CRS_Serial_UnitTest` on ranks > 0. I get output looking like:
```
***
*** Unit test suite ...
***


Sorting tests by group name then by the order they were added ... (time = 2.13e-05)

Running unit tests ...

1. CrsMatrixAdapter_CRS_Serial_UnitTest ... 
 Comparing tnzvals == enzvals ... 
 Error, tnzvals[0] = 1.01285e-267 == enzvals[0] = 0: failed!
 
 Error, tnzvals[1] = 5.91747e+250 == enzvals[1] = 0: failed!
 
 Error, tnzvals[2] = 5.72778e+250 == enzvals[2] = 0: failed!
 
 Error, tnzvals[3] = 5.72778e+250 == enzvals[3] = 0: failed!
 
 Error, tnzvals[4] = -2.05581e+177 == enzvals[4] = 0: failed!
 
 Error, tnzvals[5] = 8.45132e-227 == enzvals[5] = 0: failed!
 
 Error, tnzvals[6] = 5.53291e-222 == enzvals[6] = 0: failed!
 
 Error, tnzvals[7] = 1.16173e+282 == enzvals[7] = 0: failed!
```
Presumably, some view isn't initialized.

**Comment by bartlettroscoe** [comment]
@trilinos/framework, @sebrowne, @trilinos/amesos2, @ccober6 

FYI: This test is randomly handing about 30% of the PR iterations that run the Amesos2 tests in the build:

* `rhel7_sems-gnu-8.3.0-openmpi-1.10.7-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables`

as shown in [this query](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2023-02-01&end=2023-02-19&filtercount=3&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-gnu-8.3.0-openmpi-1.10.7-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=testname&compare3=61&value3=Amesos2_CrsMatrix_Adapter_Consistency_Tests_MPI_4) and [this query](https://trilinos-cdash.sandia.gov/queryTests.php?project=Trilinos&begin=2023-02-01&end=2023-02-19&filtercount=3&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Pull%20Request&field2=buildname&compare2=63&value2=rhel7_sems-gnu-8.3.0-openmpi-1.10.7-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables&field3=testname&compare3=61&value3=Amesos2_CrsMatrix_Adapter_Consistency_Tests_MPI_4).

It just took out my PR iteration https://github.com/trilinos/Trilinos/pull/11546#issuecomment-1436079390.

Again, this is taking out about 30% of the PR iterations that run the Amesos2 tests all by itself!

NOTE: This was reported on Jan 10 and here it is Feb 19 and this is still not fixed.

Can we at least disable this test in PR testing so it does not take out 30% of the PR build/test iterations?

**Comment by sebrowne** [comment]
@bartlettroscoe I'm fairly certain that the timeouts on your PR runs are actually more closely related to #11530.  @rppawlo this is probably that whack-a-mole thing we had discussed related to it being a problem with that newer SEMS MPI.  I will add this disable to my PR #11567 but no guarantees that this will be the last one (and disabling these tests is really not the right approach given that we're nearly certain that it's a problem with the underlying toolchain, it's just a bandaid to help alleviate the development block given that I'm currently working on nailing down what is different from the older OpenMPI 1.10.1 installation).

**Comment by sebrowne** [comment]
@cgcgcg I agree, your reported issue looks different.  That comment was only about the failure case that Ross saw (all of his showed timeouts on the dashboard), sorry I should have been more clear.

**Comment by bartlettroscoe** [comment]
The random hangs may not be related to the error reported [above](https://github.com/trilinos/Trilinos/issues/11454#issue-1528202779).  A new issue should be posted.

**Comment by sebrowne** [comment]
> The random hangs may not be related to the error reported [above](https://github.com/trilinos/Trilinos/issues/11454#issue-1528202779). A new issue should be posted.

#11530 is already filed

**Comment by ndellingwood** [comment]
@cgcgcg can you post a reproducer configuration and machine this occurred on? We have been testing nightly Cuda builds with UVM OFF but I have not seen this random fail mode you posted (only the random fails in CI that occurred after the sems module update which was resolved separately), a reproducer would be helpful

**Comment by cgcgcg** [comment]
@ndellingwood Just sent an email with details about my build.
